### DISCUSSION TITLE
Aptos Property Types

### DISCUSSION THREAD

**Discussion by memxor** [aptos-labs/aptos-developer-discussions#413] [Questions]
### Discord user ID

memxor

### Describe your question in detail.

![image](https://github.com/user-attachments/assets/d8f35387-cc31-4a62-998b-1b19d17cd616)
Is this the correct way of describing property types? Even tho I'm entering it as u64, it shows up as Hex in the Explorer and Wallet

![image](https://github.com/user-attachments/assets/86e4c8f4-6f40-4c6c-abdc-1c60b56fbf31)
How can I fix this?

### What error, if any, are you getting?

Property Values are showing up in Hex rather than Decimal

### Which operating system are you using?

Windows

### Which SDK or tool are you using? (if any)

N/A

**Comment by gregnazario** [comment]
I probably need a little more context around the code.

But, generally if you provide the type, it should be decodable.  It's possible that the explorer needs a fix here on the UI parsing side

**Comment by memxor** [comment]
hey @gregnazario thanks for your response, the code is `token::create_tokendata`, PropertyKeys, PropertyValues and PropertyTypes.
![image](https://github.com/user-attachments/assets/57d5ca02-b340-47b8-806a-fd9745859ee2)

As you see I've entered them as u64, but in Wallet (Petra) and Explorer, it shows as Hex
![image](https://github.com/user-attachments/assets/caa5fd7c-0eea-4f7d-93bc-d14880d241c3)
![image](https://github.com/user-attachments/assets/948c989d-862d-4243-a2b0-24782bb0923a)

Is my code correct?

**Reply by ajaythxkur** [reply]
maybe u need to add values in String too instead of u8 vector

`vector<String>
[
    utf8::(b"25"),
    utf8::(b"30")
    utf8::(b"27")
]
`

**Reply by memxor** [reply]
It doesn't work, it says it needs a vector of u8

![image](https://github.com/user-attachments/assets/355939c4-0607-4a68-b808-482ad7007e3d)

**Reply by gregnazario** [reply]
Ah I see, using the legacy token.

The legacy token had no rich typing, so there is no way to have the explorer know what type it is without a lot of customization in the UI.

I would suggest you take a look at the newer digital assets at 0x4
